docs/static/breaking-changes.asciidoc

```diff
@@ -1,93 +1,256 @@
 [[breaking-changes]]
 == Breaking changes
 
-This section discusses the changes that you need to be aware of when migrating your application to Logstash {version}.
+This section discusses the changes that you need to be aware of when migrating your application to Logstash {version} from the previous major release of Logstash (2.x).
+
+[float]
+=== When to Upgrade
```

Comment:
@pickypg has been driving this and has done a great job with drafting up the upgrade docs. It almost seems like a drop in replacement for our upgrade logstash section. I do agree with @dedemorton that this stuff fits more into the upgrade section than here (though we should certainly link to the breaking changes).